.github/workflows/pr.yml

```diff
@@ -6,53 +6,111 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        node: [16]
+        node:
+          - 20
     steps:
       - name: Checkout repo
         uses: actions/checkout@v3
-      - name: Use Node ${{ matrix.node }}
+      - name: "Use Node ${{ matrix.node }}"
         uses: actions/setup-node@v3
         with:
-          node-version: ${{ matrix.node }}
-          cache: "yarn"
+          node-version: "${{ matrix.node }}"
+          cache: yarn
+      - name: Setup Anchor & Build
```

Comment:
Aside: is github CI smart enough to only run certain tasks if certain files are changed? For example, we should only run anchor tasks if solana files are touched right?

Reply:
not easily. right now it will do all parts of the work flow on each run. we can improve this later. it's not too slow by comparison to how it was before.

Reply:
Conditional runs are pretty hard, but I think there are easier ways to make it less imapctful.

Two things on this if it becomes a pain point:
- We can use a docker image with rust tools and/or solana tools preinstalled.
- We can run this setup in parallel while we're doing EVM build, tests, etc, which I suspect will take longer.